### PR TITLE
Entity to dto converter, Api exception controller, user update functionality

### PR DESCRIPTION
**Pull request by IhorVrublevskyi**
Some important stuff for everyone:
- Entity to dto and back converter
- Api exception controller for simple exception handling

And user update functionality for user cabinet

### CODE REVIEW

File: src/main/java/com/softserve/edu/cajillo/config/ModelMapperConfig.java [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-
-    }
-
-    private void mapDtoToEntities(ModelMapper modelMapper) {
-
-    }
-
-}
+package com.softserve.edu.cajillo.config;
+
```
Comment (KKhrystyna) [inline src/main/java/com/softserve/edu/cajillo/config/ModelMapperConfig.java]: Why same part of code ( all classes in this commit) was deleted and created one more time???

**Review by KKhrystyna — CHANGES_REQUESTED** [review]
Maybe it would be better to rename method "updateUser"  to "updatePassword"?
Refactor USER_USERNAME_NOT_FOUND_MESSAGE = "Could not find student with name=", replace student to user.

**Comment by IhorVrublevskyi** [comment]
"rename method "updateUser" to "updatePassword""
I don`t think so. This method updates ALL information about user. Not just a password.

"replace student to user"
Fixed

**Review by KKhrystyna — COMMENTED** [review]
Basically you retrieve user by username from database and check if old password.equal to new password (!) and set the same one as new password(?!). Not sure if it is ok. Going to approve this PR, but please double check it.

**Review by KKhrystyna — APPROVED** [review]

**Review by vitaliibedeshko — APPROVED** [review]

**Pull request merged**